nautobot/project-static/js/bulk_actions.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Bulk Delete Button Enhancement
+ * 
+ * Provides progressive enhancement for bulk delete buttons.
+ * Without JavaScript: buttons work normally.
+ * With JavaScript: buttons are disabled until items are selected, with helpful tooltips
+ * 
+ * Usage:
+ * Add class 'btn-bulk-delete' to any bulk delete button with these data attributes:
+ * - data-form-id: ID of the form containing the checkbox inputs. This is required.
+ * - data-help-msg: Tooltip text when disabled (e.g., "Select a cluster to remove device from"). Fallback to "Select items to delete".
+ * - data-label-checked-single: Button text for single item (e.g., "Remove from cluster"). Fallback to button text.
+ * - data-label-checked-multiple: Button text for multiple items with [count] placeholder (e.g., "Remove from 3 clusters"). Fallback to data-label-checked-single.
+ */
+
+function initializeBulkDeleteButtons(context = document) {
+    $(context).find('.btn-bulk-delete').each(function() {
```

Comment:
Any chance you'd be willing to rewrite this in vanilla (non-jQuery) JS? We're trying to move in that direction overall.

I like this feature overall though!

Reply:
Started down that rabbit hole, ended up opening #7368, which I can take a swing at after my open PRs are finished.